File: tests/integration/benchmarks/fixtures/csharp/rosetta/src/Types/Entry.cs
namespace Rosetta.Types;

/// <summary>
/// A timestamped entry in the data pipeline.
/// </summary>
public class Entry
{
    public string Id { get; }
    public DateTime Timestamp { get; }
    public Status Status { get; }

    public Entry(string id, DateTime timestamp, Status status)
    {
        Id = id;
        Timestamp = timestamp;
        Status = status;
    }
}

File: tests/integration/benchmarks/fixtures/java/rosetta/src/com/rosetta/types/Entry.java
package com.rosetta.types;

import java.time.Instant;

/**
 * A timestamped entry in the data pipeline.
 */
public class Entry {
    private final String id;
    private final Instant timestamp;
    private final Status status;

    public Entry(String id, Instant timestamp, Status status) {
        this.id = id;
        this.timestamp = timestamp;
        this.status = status;
    }

    public String getId() { return id; }
    public Instant getTimestamp() { return timestamp; }
    public Status getStatus() { return status; }
}
